src/main/java/com/imsweb/geocoder/GeocodeOutput.java

```diff
@@ -217,358 +146,42 @@ public void setParsedAddress(Address parsedAddress) {
-                        result.setfGeometrySrid(GeocoderUtils.value(parts[106]));
-                        result.setfGeometry(GeocoderUtils.value(parts[107]));
-                        result.setfSource(GeocoderUtils.value(parts[108]));
-                        result.setfVintage(GeocoderUtils.value(parts[109]));
-                        result.setfPrimaryIdField(GeocoderUtils.value(parts[110]));
-                        result.setfPrimaryIdValue(GeocoderUtils.value(parts[111]));
-                        result.setfSecondaryIdField(GeocoderUtils.value(parts[112]));
-                        result.setfSecondaryIdValue(GeocoderUtils.value(parts[113]));
-                        result.setNaaccrCensusTractCertaintyCode(GeocoderUtils.value(parts[114]));
-                        result.setNaaccrCensusTractCertaintyName(GeocoderUtils.value(parts[115]));
-
-                        // test if there are any census tracts returned
-                        int nextPosition;
-                        if (parts.length > 151) {
-                            addCensus(result, parts, 1990, 116);
-                            addCensus(result, parts, 2000, 128);
-                            addCensus(result, parts, 2010, 140);
-                            nextPosition = 152;
-                        }
-                        else if (parts.length > 127) {
-                            addCensus(result, parts, CURRENT_CENSUS_YEAR, 116);
-                            nextPosition = 128;
-                        }
-                        else
-                            nextPosition = 116;
-
-                        if (parts.length > nextPosition + 2) {
-                            result.setMicroMatchStatus(GeocoderUtils.value(parts[nextPosition]));
-                            result.setPenaltyCode(GeocoderUtils.value(parts[nextPosition + 1]));
-                            result.setPenaltyCodeSummary(GeocoderUtils.value(parts[nextPosition + 2]));
-                        }
-                        return result;
-                    })
-                    .collect(Collectors.toList());
-        }
+        try {
+            JsonNode node = _OBJECT_MAPPER.readTree(resultString);
 
-    }
+            JsonNode data = node.get("data");
+            JsonNode versionNode = data.get("version");
+            String apiVersion = versionNode.get("major").asText() + "." + versionNode.get("minor").asText() + "." + versionNode.get("build").asText();
+
+            GeocodeOutput output = _OBJECT_MAPPER.convertValue(node.get("data"), GeocodeOutput.class);
+            output.setApiVersion(apiVersion);
+            output.setUrl(url);
+            output.setStatusCode(node.get("statusCode").asInt());
+            if (!node.get("error").isNull())
+                output.setError(node.get("error").asText());
+            output.setTransactionId(data.get("transactionGuid").asText());
+            output.setTimeTaken(data.get("transactionGuid").asDouble());
```

Comment:
There is a copy/paste error here, timeTaken is being set to transactionGuid.

Reply:
At least I got the type right...?